File: n-back-metrics.r
# Load required libraries
library(dplyr)

# Read the CSV file
data <- read.csv("n-back_results_test.csv")

# Calculate metrics
metrics <- data %>%
  summarise(
    Total = n(),
    TP = sum(ShouldRespond == 1 & Response == 1),
    TN = sum(ShouldRespond == 0 & Response == 0),
    FP = sum(ShouldRespond == 0 & Response == 1),
    FN = sum(ShouldRespond == 1 & Response == 0),
    P = sum(ShouldRespond == 1),
    N = sum(ShouldRespond == 0)
  ) %>%
  mutate(
    Accuracy = (TP + TN) / Total,
    Sensitivity = TP / P,
    Specificity = TN / N,
    ResponseDelay = mean(data$Response.Time..ms., na.rm = TRUE)
  )

# Print the results
print(metrics)

File: n-back-metrics.py
import pandas as pd
import numpy as np

# Read the CSV file
data = pd.read_csv("n-back_results_test.csv")

# Calculate metrics
metrics = pd.DataFrame({
    'Total': len(data),
    'TP': ((data['ShouldRespond'] == 1) & (data['Response'] == 1)).sum(),
    'TN': ((data['ShouldRespond'] == 0) & (data['Response'] == 0)).sum(),
    'FP': ((data['ShouldRespond'] == 0) & (data['Response'] == 1)).sum(),
    'FN': ((data['ShouldRespond'] == 1) & (data['Response'] == 0)).sum(),
    'P': data['ShouldRespond'].sum(),
    'N': (data['ShouldRespond'] == 0).sum()
}, index=[0])

# Calculate additional metrics
metrics['Accuracy'] = (metrics['TP'] + metrics['TN']) / metrics['Total']
metrics['Sensitivity'] = metrics['TP'] / metrics['P']
metrics['Specificity'] = metrics['TN'] / metrics['N']
metrics['ResponseDelay'] = data['Response Time (ms)'].mean()

# Print the results
print(metrics)